soco/events.py

```diff
@@ -136,7 +137,17 @@ def parse_event_xml(xml_event):
                     # If DIDL metadata is returned, convert it to a music
                     # library data structure
                     if value.startswith('<DIDL-Lite'):
-                        value = from_didl_string(value)[0]
+                        # Wrap any parsing exception in a SoCoFault, so the
+                        # user can handle it
+                        try:
+                            value = from_didl_string(value)[0]
+                        except SoCoException as ex:
+                            log.error("Event contains illegal metadata"
+                                      "for '%s'.\n"
+                                      "Error message: '%s'\n"
+                                      "The result will be a SoCoFault.",
+                                      tag, str(ex))
```

Comment:
I think the proper way to create a traceback as a string from a exceptions is to use the traceback module: https://stackoverflow.com/questions/4564559/get-exception-description-and-stack-trace-which-caused-an-exception-all-as-a-st

Reply:
I didn't want to log the traceback here, since the exception will be reraised. Instead, I wanted to give extra information about the tag and alert the user that a following exception might be caused by a SoCoFault.

Reply:
Ok